srtcore/tsbpd_time.h

```diff
@@ -0,0 +1,157 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+#ifndef INC_SRT_TSBPD_TIME_H
+#define INC_SRT_TSBPD_TIME_H
+
+#include "platform_sys.h"
+#include "sync.h"
+#include "utilities.h"
+
+namespace srt
+{
+
+/// @brief TimeStamp-Based Packet Delivery Mode (TSBPD) time conversion logic.
+/// Used by the receiver to calculate delivery time of data packets.
+/// See SRT RFC Section "Timestamp-Based Packet Delivery".
+class CTsbpdTime
+{
+    typedef srt::sync::steady_clock  steady_clock;
+    typedef steady_clock::time_point time_point;
+    typedef steady_clock::duration   duration;
+    typedef srt::sync::Mutex         Mutex;
+
+public:
+    CTsbpdTime()
+        : m_bTsbPdMode(false)
+        , m_tdTsbPdDelay(0)
+        , m_bTsbPdWrapCheck(false)
+    {
+    }
+
+    /// Set TimeStamp-Based Packet Delivery Mode (receiver).
+    /// @param [in] timebase local time base (uSec) of packet time stamps including buffering delay.
+    /// @param [in] wrap wrapping period.
+    /// @param [in] delay negotiated TsbPD delay (buffering latency).
+    void setTsbPdMode(const time_point& timebase, bool wrap, duration delay);
+
+    /// @brief Check if TSBPD logic is enabled.
+    /// @return true if TSBPD is enabled.
+    bool isEnabled() const { return m_bTsbPdMode; }
+
+    /// @brief Apply new state derived from other members of a socket group.
+    /// @param timebase TSBPD base time.
+    /// @param wrp wrap period (enabled or not).
+    /// @param delay TSBPD delay.
+    /// @param udrift clock drift.
+    void applyGroupTime(const time_point& timebase, bool wrp, uint32_t delay, const duration& udrift);
+
+    /// @brief Apply new clock state (TSBPD base and drift) derived from other members of a socket group.
+    /// @param timebase TSBPD base time.
+    /// @param wrp the state of wrapping period (enabled or disabled).
+    /// @param udrift Clock drift.
+    void applyGroupDrift(const time_point& timebase, bool wrp, const duration& udrift);
+
+    /// @brief Add new drift sample from an ACK-ACKACK pair.
+    /// ACKACK pakcets are sent immideatly (except for UDP buffering).
+    /// 
+    /// @param [in] pktTimestamp Timestamp of the arrived ACKACK packet.
+    /// @param [out] w_udrift Current clock drift value
+    /// @param [out] w_newtimebase Current TSBPD base time
+    /// 
+    /// @return true if TSBPD base time was changed, false otherwise.
+    bool addDriftSample(uint32_t                  pktTimestamp,
+                        steady_clock::duration&   w_udrift,
+                        steady_clock::time_point& w_newtimebase);
+
+    /// @brief Get packet TSBPD time with buffering delay.
+    /// The target time when to deliver the packet to an upstream application.
+    /// @param [in] usPktTimestamp 32-bit value of packet timestamp field (microseconds).
+    ///
+    /// @returns Packet TSBPD base time with buffering delay.
```

Comment:
@return or @returns? Should this be consistent throughout? Because it is not.

Reply:
[Both are acceptable](https://www.doxygen.nl/manual/commands.html#cmdreturns).
But you are right, to be consistent, better to use a single one, e.g. `@return`.